### DISCUSSION TITLE
onlyM wont even start

### DISCUSSION THREAD

**Discussion by nickprosser1966** [AntonyCorbett/OnlyM#472] [Q&A]
I recently got a new PC running win 11. I installed OnlyM and it ran fine, did a meeting with all media no problem.

a few days later, i had possibly updated the pc via MS, installed a few programs . set pc to not use onedrive(i hate it). the pc still runs fine
But OnlyM wont run. after first install the PC asks for permission, i click yes and it thinks about it but nothing happens. then subsequent attempts to run it just flatly doesnt run it. I have unistalled and reinstalled a few times . I have disabled the built in virus protection and firewalls and no difference. I have installed .net5 after reading discussion here, no difference. I have been through the uninstall and then purged the pc of as many files as i can find and reinstalled , still wont run.  cannot find any log file in documents or on hard drive. am now at a loss 
can you help please?

**Comment by AntonyCorbett** [comment]
@nickprosser1966 Suggested procedure:

1. Delete the following folder (search for this in File Explorer): %AppData%\OnlyM
2. Uninstall OnlyM
3. Download the latest pre-release (2.1.0.2) from https://github.com/AntonyCorbett/OnlyM/releases
4. Install and try again

Regards
Antony

**Reply by nickprosser1966** [reply]
Extensive search cannot find that folder  hidden files showing as well. any ideas where it "should" be? 
uninstall and reinstall of 2.1.0.2 same results. even specified a different installation folder
Nick

**Reply by AntonyCorbett** [reply]
Regarding the "%AppData%\OnlyM" path, specify it in the search bar of File Explorer and press ENTER. The macro will expand to the path.

**Reply by nickprosser1966** [reply]
![onlymsearch](https://user-images.githubusercontent.com/120631362/207858465-5bc897d9-31e7-4dfd-8a1a-8fb4cf1c1640.jpg)
i dont know what im doing wrong, search finds no match. sorry to be a pain

**Reply by AntonyCorbett** [reply]
I should have explained better :)

![image](https://user-images.githubusercontent.com/24551146/207891717-490ec7b6-fe09-484f-af7b-90a8e7f63657.png)

**Comment by nickprosser1966** [comment]
![Screenshot 2022-12-16 074134](https://user-images.githubusercontent.com/120631362/208048300-ae489379-f31f-4a8b-9784-2af5aead7cbd.jpg)
<shrug>

**Comment by AntonyCorbett** [comment]
@nickprosser1966 Have you turned on "Controlled folder access" in Windows? If so, then OnlyM won't be able to create that folder and could explain the behaviour.

**Reply by nickprosser1966** [reply]
Previously i had tried with all inbuilt virus protection off , install and run. 
following you recent reply this setting was on. I have switched it off , reinstalled etc .still the same OnlyM wont start.
Im sorry to keep bothering you with this as it did run perfectly fine but something I have done has stopped it working. 
Nick

**Reply by AntonyCorbett** [reply]
Does Windows Event Viewer show any errors when you attempt to run the application? i.e. under Windows Logs => Application

**Reply by nickprosser1966** [reply]
The description for Event ID 0 from source Application cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.

If the event originated on another computer, the display information had to be saved with the event.

The following information was included with the event: 

Access to the path 'C:\Program Files (x86)\OnlyM2\OnlyM\BrowserCache' is denied.

The message resource is present but the message was not found in the message table

**Reply by nickprosser1966** [reply]
that sub folder \OnlyM2\Onlym\ doesnt seem to exist
(OnlyM2 is instalation directory chosen at install to see if it solved problem)

**Reply by AntonyCorbett** [reply]
This would seem to indicate a problem with access rights to the installation folder. If you are sure that "Controlled folder access" is turned off, please check "Advanced app settings" and confirm that you can install apps from "Anywhere"

**Reply by nickprosser1966** [reply]
controlled folder access is off , advanced app settings is "anywhere" . i have been through the folder permissions and allowed full access. i have run the installer in administrator mode and the program. still no joy . slide manager runs fine.

latest event error is as follows

The description for Event ID 0 from source Application cannot be found. Either the component that raises this event is not installed on your local computer or the installation is corrupted. You can install or repair the component on the local computer.

If the event originated on another computer, the display information had to be saved with the event.

The following information was included with the event: 

CefSettings.CachePath now requires an absolute path, the path provided is non-absolute. You can use System.IO.Path.GetFullPath to obtain an absolute path. Current value:OnlyM\BrowserCache

The message resource is present but the message was not found in the message table

**Reply by nickprosser1966** [reply]
this warning is in System logs which i think relates to me installing OnlyM


The machine-default permission settings do not grant Local Activation permission for the COM Server application with CLSID 
{C2F03A33-21F5-47FA-B4BB-156362A2F239}
 and APPID 
{316CDED5-E4AE-4B15-9113-7055D84DCC97}
 to the user NT AUTHORITY\LOCAL SERVICE SID (S-1-5-19) from address LocalHost (Using LRPC) running in the application container Unavailable SID (Unavailable). This security permission can be modified using the Component Services administrative tool.

**Reply by AntonyCorbett** [reply]
> this warning is in System logs which i think relates to me installing OnlyM
> 
> The machine-default permission settings do not grant Local Activation permission for the COM Server application with CLSID {C2F03A33-21F5-47FA-B4BB-156362A2F239} and APPID {316CDED5-E4AE-4B15-9113-7055D84DCC97} to the user NT AUTHORITY\LOCAL SERVICE SID (S-1-5-19) from address LocalHost (Using LRPC) running in the application container Unavailable SID (Unavailable). This security permission can be modified using the Component Services administrative tool.

This can safely be ignored (google if you want more info)

**Reply by AntonyCorbett** [reply]
> CefSettings.CachePath now requires an absolute path, the path provided is non-absolute. You can use System.IO.Path.GetFullPath to obtain an absolute path. Current value:OnlyM\BrowserCache

This is significant. Please would you right-click the "Documents" folder in File Explorer, select "Properties" and report its "Location".

**Reply by nickprosser1966** [reply]
C:\Users\nick_

**Reply by AntonyCorbett** [reply]
I wonder if your user account has access rights to that folder? If you navigate to the "c:\users\nick_" folder in File Explorer, right click and select New => Folder, does it allow you to create a sub-folder without having to elevate to admin?

**Reply by nickprosser1966** [reply]
Yes. And the account I use says its Administrator level. I can click on security messages that says "requires administrator access" and it then progresses (on other things ,not this problem) so I believe the account has sufficient rights. I have run the install using "run as adminitrator" and the actual program as well.

**Reply by nickprosser1966** [reply]
do you think a "repair " to windows operating system might help? its so annoying as OnlyM did work initially

**Reply by AntonyCorbett** [reply]
>  I can click on security messages that says "requires administrator access" and it then progresses

Ah, that's the problem. Your documents folder should _not_ require administrative access rights! My guess is that when you removed OneDrive you had to recreate a Documents folder and did so using elevated access. The Documents folder is user-specific and the user should have full access to it without needing to elevate.

**Reply by nickprosser1966** [reply]
No, I can create a folder  without any restrictions. and sub folders without any messages.
the rest was just an addendum. sorry.

**Reply by AntonyCorbett** [reply]
What is the location of the Documents folder that appears as a link in the left panel of File Explorer? Right-click "Documents" and select "Properties" to view the Location.

**Reply by nickprosser1966** [reply]
C:\Users\nick_

**Reply by AntonyCorbett** [reply]
Can you create a subfolder in the c:\Users\nick_\Documents folder (without requiring elevation)?

**Reply by nickprosser1966** [reply]
Yes

**Reply by AntonyCorbett** [reply]
Please open a PowerShell window and type the following command to retrieve the "Documents" path:

`[Environment]::GetFolderPath("MyDocuments")`

**Reply by nickprosser1966** [reply]
![POWERSHELL01](https://user-images.githubusercontent.com/120631362/209151965-7901e02a-5947-4e54-93e8-714b341665c2.jpg)

Im not that familiar with powershell. having pasted your command in it doesnt appear to do anything. even navigating to the root dir with CD.. command and entering that command doesnt do anything...unless im missing something. sorry

**Reply by scottys94** [reply]
@nickprosser1966 Just for testing, can you create another Windows user account with Administrator rights, log into it, then see if OnlyM will run?

**Reply by AntonyCorbett** [reply]
@nickprosser1966 the powershell command shows that you have no "documents" folder known to Windows. I'm not sure how best to fix that, but it is likely to affect other applications too. A repair of Windows may work. You can also try @scottys94's suggestion above.

**Reply by nickprosser1966** [reply]
Well setting up a new account worked. I will probably transfer use to the new account and delete the old account eventually.

Thank you for your patience. I really appreciate all the help you have given.

It seems this all stemmed from me deleting Onedrive.

Nick

**Reply by AntonyCorbett** [reply]
@nickprosser1966 excellent! @scottys94 usually steps in with the answer when he sees me struggling :)
